snooty/test_parser.py

```diff
@@ -87,6 +84,58 @@ def test_tabs() -> None:
     assert isinstance(diagnostics[1], UnknownTabset)
     assert isinstance(diagnostics[2], DocUtilsParseError)
 
+    # Test a tab with no tabid
+    parser = rstparser.Parser(project_config, JSONVisitor)
+    page, diagnostics = parse_rst(
+        parser,
+        tabs_path,
+        """
+.. tabs-drivers::
+
+   .. tab::
+      :tabid: python
+
+      Python!
+
+   .. tab::
+
+      Tab with no ID
+
+   .. tab::
+      :tabid: javascript
+
+      Javascript!
+
+   .. tab::
+      :tabid: php
+
+      PHP!
+""",
+    )
+    page.finish(diagnostics)
+    assert [(type(d), d.severity) for d in diagnostics] == [
+        (DocUtilsParseError, Diagnostic.Level.error),
+        (UnknownTabID, Diagnostic.Level.error),
+    ]
+
+    check_ast_testing_string(
+        page.ast,
+        """
+<root fileid="test_tabs.rst">
+    <directive name="tabs" tabset="drivers">
+        <directive name="tab" tabid="python">
+            <text>Python</text>
+            <paragraph><text>Python!</text></paragraph>
+        </directive>
+        <directive name="tab" tabid="php">
```

Comment:
Should there be a tab for JS approximately here?

Reply:
Nope: I accidentally made the test confusing. The language is called nodejs; there is no `javascript` language defined in the tabset

Reply:
OH! That actually makes a lot of sense, and I should read more carefully :D

Reply:
It was a good thing to clarify! :D